File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/secret/impl/PinPatternSecret.java
package at.tugraz.tc.cyfile.secret.impl;

import com.andrognito.patternlockview.PatternLockView;

import java.util.List;

public class PinPatternSecret extends AbstractSecret {

    private List<PatternLockView.Dot> dotList;

    private String patterns;

    public PinPatternSecret(List<PatternLockView.Dot> pattern) {
        this.dotList = pattern;
    }

    public PinPatternSecret(String patterns) {
        this.patterns = patterns;
    }

    @Override
    public String getSecretValue() {
        if (patterns != null) {
            return patterns;
        }

        if (dotList != null) {
            StringBuilder str = new StringBuilder();

            for (PatternLockView.Dot dot :
                    dotList) {
                str.append(dot.getRow()).append(dot.getColumn());
            }
            return str.toString();
        }

        return null;
    }


}

File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/secret/impl/PinPatternSecret.kt
package at.tugraz.tc.cyfile.secret.impl

import com.andrognito.patternlockview.PatternLockView

class PinPatternSecret : AbstractSecret {

    private val patterns: String

    override val secretValue: String
        get() {
            return patterns
        }

    constructor(pattern: List<PatternLockView.Dot>) {
        this.patterns = convert(pattern)
    }

    constructor(patterns: String) {
        this.patterns = patterns
    }

    private fun convert(pattern: List<PatternLockView.Dot>): String {
        val str = StringBuilder()
        pattern
                .forEach { x -> str.append(x.row).append(x.column) }
        return str.toString()
    }


}
